ndk/src/events/geocoded.ts

```diff
@@ -0,0 +1,649 @@
+};
+
+/**
+ * 
+ * This class represents a geocoded Nostr event.
+ * 
+ * @author sandwich.farm
+ * @extends NDKEvent
+ * 
+ */
+export class NDKEventGeoCoded extends NDKEvent {
+
+    private static readonly EARTH_RADIUS: number = 6371; // km
+    private static readonly GEOHASH_PRECISION: number = 12;
+    private static readonly BASE32: string = '0123456789bcdefghjkmnpqrstuvwxyz';  
+    private static readonly geohashFilterFn = (tag: NDKTag) => tag[0] === 'g' && (tag[2] === "gh" || tag[2] === "geohash" || tag.length === 2); //`g` tags with a length of 2 are NIP-52 geohashes
+
+    private _dd: DD | undefined;
+
+    constructor( ndk: NDK | undefined, rawEvent?: NostrEvent ) {
+        super(ndk, rawEvent);
+        if(NDKEventGeoCoded.isGeohash(this.geohash)){
+            this.dd = this.geohash;
+        }
+    }
+
+    static from(event: NDKEvent): NDKEventGeoCoded {
+        return new NDKEventGeoCoded(event.ndk, event.rawEvent());
+    }
+
+    /**
+     * Compares two geohashes and sorts them based on their length in descending order.
+     * 
+     * @param a - The first geohash to compare.
+     * @param b - The second geohash to compare.
+     * @returns A negative number if `b` is longer than `a`, a positive number if `a` is longer than `b`,
+     *          or zero if they have the same length.
+     * 
+     * @static
+     */
+    static sortGeohashesFn(a: string, b: string): number {
+        return b.length - a.length;
+    }
+
+    /**
+     * Generates a list of filterable geohashes from a given full geohash.
+     * This method creates a set of progressively shorter prefixes of the provided geohash,
+     * each representing a broader geographic area. This set can be used for filtering or
+     * querying geospatial data structures and is sorted from the most specific (longest)
+     * to the least specific (shortest).
+     * 
+     * @param fullGeohash - The full geohash string from which to generate filterable prefixes.
+     * @returns An array of geohash strings, each a prefix of the full geohash, sorted by descending length.
+     * 
+     * @static
+     */
+    static generateFilterableGeohash(fullGeohash: Geohash): Geohash[] {
+        const geohashes: Set<Geohash> = new Set();
+        const deref: Geohash = String(fullGeohash); 
+        for (let i = String(fullGeohash).length; i > 0; i--) {
+            const n = deref.substring(0, i);
+            geohashes.add(n);
+        }
+        return Array.from(geohashes);
+    }
+
+    /**
+     * Sets the geographical coordinates or geohash for the current instance, updating latitude, longitude, and geohash values accordingly.
+     * If a geohash string is provided, it decodes it into latitude and longitude. If coordinates are provided, it sets those directly and
+     * generates a corresponding geohash.
+     * 
+     * @param coords The coordinates or geohash string to set. Can be undefined to perform no operation.
+     */
+    set dd(coords: Coords | undefined) {
```

Comment:
what's `dd`?

Reply:
`dd` stands for Decimal degrees, it's the standard terminology for lat/lon coordinates.